specs/HubProtocol.md

```diff
@@ -348,8 +363,160 @@ World;0:;
 
 Note that the final frame still ends with the `;` terminator, and that since the body may contain `;`, newlines, etc., the length is specified in order to know exactly where the body ends.
 
+## MessagePack (MsgPack) encoding
+
+In the MsgPack Encoding of the SignalR Protocol, each Message is represented as a single MsgPack array containing items that correspond to properties of the given hub protocol message. The array items may be primitive values, arrays (e.g. method arguments) or objects (e.g. argument value). The first item in the array is the message type.
```

Comment:
Yes, this is one of the representations of objects in MsgPack. Previously, I had only values which worked well in C# but was hard to handle in JS (and also see this [comment](https://github.com/aspnet/SignalR/pull/587#issuecomment-311338633)). You can't convert this directly to object because you don't have fields names - so yes it will be an array. A different way of doing this is to send an object (i.e. including field names etc.) serialized to MsgPack format. I did it originally and the messages were much bigger than doing arrays - so there was not a real benefit of using message pack. I also don't think it is needed to send hub protocol messages as objects since we control the protocol.